components/lib/dialog/Dialog.js

```diff
@@ -329,27 +326,63 @@ export const Dialog = React.forwardRef((inProps, ref) => {
 
     const enableDocumentSettings = () => {
         bindGlobalListeners();
-
-        if (props.blockScroll || (props.maximizable && maximized)) {
-            DomHandler.addClass(document.body, 'p-overflow-hidden');
-        }
+        updateGlobalDialogsRegistry(true)
     };
 
     const disableDocumentSettings = () => {
         unbindGlobalListeners();
+        updateGlobalDialogsRegistry(false)
+    };
 
-        const isMaximized = props.maximizable && maximized;
+    const updateScrollBlocker = () => {
+        // Scroll should be unblocked if there is at least one dialog that blocks scrolling:
+        const isThereAnyDialogThatBlocksScrolling = (
+            document.primeDialogParams &&
+            document.primeDialogParams.some(i => i.hasBlockScroll)
+        )
 
-        if (props.modal) {
-            const hasBlockScroll = props.blockScroll || (document.primeDialogParams && document.primeDialogParams.some((param) => param.hasBlockScroll));
+        if (isThereAnyDialogThatBlocksScrolling) {
+            DomHandler.addClass(document.body, 'p-overflow-hidden')
+        }
+        else {
+            DomHandler.removeClass(document.body, 'p-overflow-hidden')
+        }
+    }
+
+    const updateGlobalDialogsRegistry = (isMounted) => {
+        // Update current dialog info in global registry if it is mounted:
+        if (isMounted) {
+            const newParam = { id: idState, hasBlockScroll: shouldBlockScroll };
 
-            if (hasBlockScroll || isMaximized) {
-                DomHandler.removeClass(document.body, 'p-overflow-hidden');
+            // Create registry if not yet created:
+            if (!document.primeDialogParams) {
+                document.primeDialogParams = []
+            }
+
+            const currentDialogIndexInRegistry = document.primeDialogParams.findIndex(
+                dialogInRegistry => dialogInRegistry.id === idState
+            );
+
+            if (currentDialogIndexInRegistry === -1) {
+                document.primeDialogParams = [...document.primeDialogParams, newParam]
+            }
+            else {
+                // Is it OK to mutate this array?
+                // If mutation is not desired - try to use modern
+                // `.toSpliced` method
+                // (see here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSpliced)
+                document.primeDialogParams[currentDialogIndexInRegistry] = newParam
```

Comment:
We are only supporting modern browsers so I think its OK for you to use toSpliced if its the correct thing to do?